File: HW02-Fortran/pi.f90
program pi

   ! This program estimates pi
   implicit none  ! Disables the auto type ijk....

   ! Type declarations
   real :: x,y,dist
   integer :: i, in_count, total
   
   total = 100000000

   ! Executable statements
   in_count = 0
   i = total
   do while (i > 0)
      x = RAND(0) - 0.5
      y = RAND(0) - 0.5
      dist = sqrt(x * x + y * y)
      if (dist < 0.5) then
         in_count = in_count + 1
      end if
      i = i - 1
   end do
   print *, "in: ", in_count, 'total', total
   print *, "pi: ", REAL(in_count) * 4.0 / total
end program pi

File: HW02-Fortran/pi.c
/*
Name: Paul Talaga
Date: Aug 10, 2020
Desc: Estimates pi by random number generation.

*/
#include <stdio.h> 
#include <stdlib.h> 
#include <math.h>


#define TOTAL 100000000

int main(){
    unsigned i;
    float x, y, d;
    unsigned in_count = 0;
    for(i = 0; i < TOTAL; i++){
        x = (float)rand() / RAND_MAX - 0.5;
        y = (float)rand() / RAND_MAX - 0.5;
        
        //d = sqrt(x*x + y*y);
        d = x*x + y*y;
        //if(d < 0.5){
        if(d < 0.25){
            in_count++;
        }
    }
    printf("pi: %f \n", (float)in_count * 4.0 / TOTAL);
    return 0;
}